tests/Mono.Android-Tests/jni/reuse-threads.c

```diff
@@ -200,3 +200,46 @@ rt_invoke_callback_on_new_thread (CB cb)
 	return 0;
 }
 
+static int
+_register_type_from_new_thread (const char *java_type_name, jobject class_loader)
+{
+	JNIEnv *env              = _get_env ("_register_type_from_new_thread");
+	jclass ClassLoader_class = (*env)->FindClass (env, "java/lang/ClassLoader");
+	jmethodID loadClass      = (*env)->GetMethodID (env, ClassLoader_class, "loadClass", "(Ljava/lang/String;)Ljava/lang/Class;");
+	jobject loaded_class     = (*env)->CallObjectMethod (env, class_loader, loadClass, java_type_name);
+
+	if ((*env)->ExceptionOccurred (env) != NULL) {
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: class '%s' cannot be loaded, Java exception thrown!", java_type_name);
+		return -1;
+	}
+
+	jmethodID Object_ctor    = (*env)->GetMethodID (env, loaded_class, "<init>", "()V");
+	jobject   instance       = (*env)->NewObject (env, loaded_class, Object_ctor);
+
+	if ((*env)->ExceptionOccurred (env) != NULL || instance == NULL) {
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: instance of class '%s' wasn't created!", java_type_name);
+		return -1;
+	}
+
+	return 0;
+}
+
+JNIEXPORT int JNICALL
+rt_register_type_on_new_thread (const char *java_type_name)
+{
+	JNIEnv *env = _get_env ("rt_register_type_on_new_thread");
+	pthread_t t;
+
+	/* 1: Create a thread... */
+	int r = pthread_create (&t, NULL, _register_type_from_new_thread, java_type_name);
```

Comment:
Kinda surprised that this compiles without warnings (assuming it *does* compile without warnings), as `pthread_create()` specifies a `void*` return value, and `int` is not implicitly convertible to `void*`.

Reply:
The older code there did the same, I didn't check the manpage. I guess it's always worth checking... :)

Reply:
hm, no, just checked the manpage:

```
SYNOPSIS
       #include <pthread.h>

       int pthread_create(pthread_t *restrict thread,
                          const pthread_attr_t *restrict attr,
                          void *(*start_routine)(void *),
                          void *restrict arg);

```